src/compiler/reviews.md

```diff
@@ -1,4 +1,156 @@
+  We expect the mechanisms and principles we articulated above to be respected by ALL reviewers, whatever their employer.
+  Does the PR concisely describe the changes that are being made?
+  Does it give a clear, transparent rationale for why the changes make sense
+  so that contributors down the line can follow the reasoning and reconstruct what's going on?
+  Have points of contention been discussed and cleared up?
+  Then you are in the clear.
+
+  If you are in doubt if something is contentious, give a heads up to `@rust-lang/compiler-contributors` and ask for another opinion.
+  If the proposed change is large and/or potentially has a big impact, create a [major change proposal][mcp].
+
+* ### Reviewing and Mentoring
+  In the course of mentoring someone through a PR it often happens that the reviewer has ended up effectively co-writing the changes.
+  This can be a tricky case because the reviewer is effectively approving their own changes.
+  There are a number of considerations to take into account when deciding how to proceed:
+
+  - If the general direction of the changes has already been approved as part of an MCP and the concrete advice given
+    during mentoring was only concerned with resolving minor technical issues, then no further review is required.
+  - Similarly, if any contentious decisions have visibly been discussed and resolved on the PR with other
+    compiler team contributors and the rest of the changes don't deviate from the general direction that has been
+    agreed upon then no further review is required either.
+  - If the PR was opened as a response to a concrete suggestion by the reviewer (and the changes are not entirely trivial)
+    then it is advisable that the final review is done by someone else.
+    However, the initial reviewer/mentor is encouraged to help bring the PR into good shape before handing it off.
+
+  In general, it is advisable to ask for a second opinion by someone knowledgable in the field in such cases,
+  just to increase the chance of uncovering oversights and blindspots a mentor might have.
+
+* ### Nobody understands the code that's being changed
+  Sometimes there is a bug in some code that nobody understands anymore.
+  The original authors are unavailable and it is hard to gauge the implications of a proposed fix.
+  In such a case it is a good idea for reviewers to nominate the PR (by tagging it with `I-nominated`)
+  in order to get it in front of more eyes during the compiler team's triage meeting.
+  In such cases it is also especially valuable to gather and document as much information as possible on the issue,
+  such as a description of the problem being fixed, points of unclarity, potential risks,
+  alternatives that have been considered, et cetera.
+  Reviewers should ask PR authors to add this kind of information as comments in the code
+  and/or to the PR message (which will become part of the git commit history).
```

Comment:
@Mark-Simulacrum, I think you wanted to discuss the wording here. Do you have specific comments?

Reply:
I wanted to suggest a few changes:

* Replace the asking on the Zulip stream with some kind of more transactional process (e.g., handover to Felix/Wesley for re-assignment -- something that more fully moves things out of the current reviewer's hands). Maybe indicating the PR is S-waiting-on-team. Questions on Zulip can easily get lost I think, and bouncing around seems ungreat. Maybe we can I-nominate for T-compiler meetings with the explicit goal of assigning a reviewer, but those meetings don't usually have full attendance so it may not be the best venue either.

Reply:
Yes, that sounds like a good idea. Maybe:

> In such a case it is a good idea for reviewers to I-nominate the PR (if they intend to stay the main reviewer) or assign a compiler team lead to the issue and add the S-waiting-on-team label. In both cases, the PR will be brought in the weekly triage meeting.

@apiraino, what do you think of that?

Reply:
I've gone ahead and updated the text with the above version.